### PR TITLE
Make showProgress viewId optional

### PR DESCRIPTION
**Pull request by dotNomad**
PR feedback follow-up. Moves the `viewId` param for `showProgress` to make it optional.

### CODE REVIEW

**Review by sagerb — APPROVED** [review]
I have mixed feelings about this change. I liked the fact that you had your think about where the progress indicator was best shown, vs doing it as an afterthought (which is how I often think of optional parameters). 

We did have it optional in prior art, so I guess I'm ok with this change. The existing calls to this helper function all provide the viewId, so the impact here is really only on new code.

I'll approve it, but when you have a chance, let's discuss a bit.

Thanks!

**Comment by dotNomad** [comment]
> I have mixed feelings about this change. I liked the fact that you had your think about where the progress indicator was best shown, vs doing it as an afterthought (which is how I often think of optional parameters).

It was technically already option since it was typed as `viewId: string | undefined`. I'm totally fine requiring a `viewId` in the function calls, and think that would provide a nice benefit.

**Comment by sagerb** [comment]
> It was technically already option since it was typed as `viewId: string | undefined`. I'm totally fine requiring a `viewId` in the function calls, and think that would provide a nice benefit.

Glad to hear it. Yes, let's require the `viewId` in the function calls then.

**Review by sagerb — CHANGES_REQUESTED** [review]
We agreed to make the change, so let's make the change within this PR.

I can make these changes for you, since I know you're busy with user testing.

**Comment by sagerb** [comment]
I've pushed a commit which removes the optionality of the viewID within Show Progress. https://github.com/posit-dev/publisher/pull/1979/commits/08ef450219aade47983a3964f60905f086e3af30

@dotNomad Perhaps you could review those changes?

**Review by sagerb — APPROVED** [review]
@dotNomad code looks good.. Would like to have him review the couple of commits which I contributed before it is merged.

File: extensions/vscode/src/views/credentials.ts
Comment (dotNomad) [inline extensions/vscode/src/views/credentials.ts:101]: Should this be `Views.Credentials`?

**Review by dotNomad — COMMENTED** [review]
I think I caught one incorrect reference.

File: extensions/vscode/src/views/credentials.ts
Comment (sagerb) [reply extensions/vscode/src/views/credentials.ts]: Yes. I've pushed an update.

**Pull request merged**
